lib/pwwka/error_handlers/chain.rb

```diff
@@ -13,18 +13,23 @@ def handle_error(message_handler_klass,receiver,queue_name,payload,delivery_info
           @error_handlers.unshift(message_handler_klass.send(:error_handler))
         end
         @error_handlers.reduce(true) { |keep_going,error_handler|
-          logf "%{error_handler_class} is being evaluated as part of pwwka's error-handling chain", error_handler_class: error_handler
-          if keep_going
-            keep_going = error_handler.new.handle_error(receiver,queue_name,payload,delivery_info,exception)
+          begin
+            logf "%{error_handler_class} is being evaluated as part of pwwka's error-handling chain", error_handler_class: error_handler
             if keep_going
-              logf "%{error_handler_class} has asked to continue pwwka's error-handling chain", error_handler_class: error_handler
+              keep_going = error_handler.new.handle_error(receiver,queue_name,payload,delivery_info,exception)
+              if keep_going
+                logf "%{error_handler_class} has asked to continue pwwka's error-handling chain", error_handler_class: error_handler
+              else
+                logf "%{error_handler_class} has halted pwwka's error-handling chain", error_handler_class: error_handler
+              end
             else
-              logf "%{error_handler_class} has halted pwwka's error-handling chain", error_handler_class: error_handler
+              logf "Skipping %{error_handler_class} as we were asked to abort pwwka's error-handling chain", error_handler_class: error_handler
             end
-          else
-            logf "Skipping %{error_handler_class} as we were asked to abort pwwka's error-handling chain", error_handler_class: error_handler
+            keep_going
+          rescue Exception => exception
+            logf "'%{error_handler_class}' failed with exception '%{exception}'", at: :fatal, error_handler_class: error_handler, exception: exception
+            abort
```

Comment:
Should this return `false` instead of exiting the process?

Reply:
Yes, I'm happy to make that happen. The reason for the abort was due to your earlier suggestion here, https://github.com/stitchfix/pwwka/pull/61#issuecomment-342140827 .

Reply:
Hmm, now I'm third-guessing myself.  Let's go with my first instinct and maybe have the log message on line 30 say something like "aborting…" or whatever

Reply:
On it.

Reply:
Done in ac386ebf4ab16ac9369819b2dd2cb551bccb645a.